pkg/reloader/collector.go

```diff
@@ -0,0 +1,121 @@
+// Copyright © 2023 Cisco
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package reloader
+
+import (
+	"regexp"
+	"strings"
+	"sync"
+
+	"golang.org/x/exp/slices"
+	corev1 "k8s.io/api/core/v1"
+
+	"github.com/bank-vaults/vault-secrets-reloader/internal/injector"
+)
+
+type Workload struct {
+	name      string
+	namespace string
+	kind      string
+}
+
+type WorkloadSecrets struct {
+	sync.Mutex
+	m map[Workload][]string
+}
+
+func NewWorkloadSecrets() *WorkloadSecrets {
```

Comment:
There is no need to use this in `main.go` since it can be created locally by the controller which would allow making the data structure fully private and used only in the context of the controller/reloader.
IMO there is no need to move reloader to internal package since it is coupled a lot with the controller logic.